doc/etc-and-usr-etc.md

```diff
@@ -0,0 +1,75 @@
+# Adding Support for the `/etc` + `/usr/etc` Layout
+
+## About
+
+This document tries to summarize our findings about the proposal to split the configuration files
+into `/usr/etc` (vendor) and `/etc` (user) directories. As you may know, YaST reads and writes
+information to many files under `/etc`, so we need to find a way to cope with this (future) layout.
+
+## How to Read the Configuration
+
+In the future, it is expected that vendor configuration files live under `/usr/etc` and user
+configuration is placed in `/etc`. Given a `example` application, the rules to determine the
+configuration are:
+
+* If `/etc/example.conf` does not exist, read `/usr/etc/example.conf`, `/usr/etc/example.d/*.conf`
+  and, finally, `/etc/example.d/*.conf`. The latter has precedence.
+* If `/etc/example.conf` does exist, just ignore the configuration under `/usr/etc` and consider
+  `/etc/example.conf` and `/etc/example.d/*.conf` only.
+
+    YaST will merge settings from those files.
+
+## Impact in YaST
+
+When it comes to reading or writing configuration files, YaST uses mainly two different mechanisms:
+
+* The new [config_files_api](https://github.com/config-files-api/config_files_api) (a.k.a. CFA) API.
+* The [good old agents](https://github.com/yast/yast-core/), which are spread through all YaST
+  codebase (search for `.scr`) files.
+
+### Extending CFA
+
+CFA offers an object-oriented way to read and write configuration files and, nowadays, it is used
+for `zypp.conf`, `chrony.conf`, `ifroute-*` and `ifcfg-*` files, etc. CFA is built around these abstractions:
+
+* **File handlers* provide access to files. By default, it simply uses the `File` class, but it can be replaced
+  with other mechanisms to allow, e.g., accessing over the network. See the discussion about supporting agents
+  to find another use case.
```

Comment:
I would mention here the main extension done in YaST which is support for configuration in chroot - https://github.com/yast/yast-yast2/blob/master/library/system/src/lib/yast2/target_file.rb#L51